File: eAutoShop/eAutoShop.UI/eautoshop_desktop/lib/models/employee_task/employee_task.dart
import 'package:json_annotation/json_annotation.dart';

part 'employee_task.g.dart';

@JsonSerializable()
class EmployeeTask {
  final int id;
  final String title;
  final String description;
  final int employeeId;
  final String employeeName;
  final int createdById;
  final String createdByName;
  final DateTime createdAt;
  final DateTime? dueDate;
  final DateTime? completedAt;
  final String state;

  const EmployeeTask({
    required this.id,
    required this.title,
    required this.description,
    required this.employeeId,
    required this.employeeName,
    required this.createdById,
    required this.createdByName,
    required this.createdAt,
    this.dueDate,
    this.completedAt,
    required this.state,
  });

  factory EmployeeTask.fromJson(Map<String, dynamic> json) =>
      _$EmployeeTaskFromJson(json);

  Map<String, dynamic> toJson() => _$EmployeeTaskToJson(this);
}

File: eAutoShop/eAutoShop.Services/Database/EmployeeTask.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eAutoShop.Services.Database;

public partial class EmployeeTask
{
    public int Id { get; set; }

    public string Title { get; set; } = null!;

    public string Description { get; set; } = null!;

    public int EmployeeId { get; set; }

    public int CreatedById { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? DueDate { get; set; }

    public DateTime? CompletedAt { get; set; }

    public string State { get; set; } = null!;

    public virtual User Employee { get; set; } = null!;

    public virtual User CreatedBy { get; set; } = null!;
}
